src/core/layout/qgslayoutitem.h

```diff
@@ -292,12 +293,17 @@ class CORE_EXPORT QgsLayoutItem : public QgsLayoutObject, public QGraphicsRectIt
     virtual void cleanup();
 
     /**
-     * Return unique graphics item type identifier.
+     * Returns a unique graphics item type identifier.
      *
      * Plugin based subclasses should return an identifier greater than QgsLayoutItemRegistry::PluginItem.
      */
     int type() const override;
 
+    /**
+     * Returns the item's icon.
+     */
+    virtual QIcon icon() { return QIcon(); }
```

Comment:
const

Reply:
Should we provide a default here? (e.g. rectangle shape icon). It's always rather ugly when a list has missing icons, and when Python code starts adding extra item types I can see them skipping the icon...